### PR TITLE
Add usedevelop flag for coverage-tox issue

### LINKED ISSUE #269: coverage: Investigate why coverage is reported to be 0%
### Steps to Reproduce

See See https://codecov.io/gh/NeurodataWithoutBorders/pynwb

### PR DESCRIPTION
**Pull request by dorukozturk**
## Motivation

Fixes #269 

Coverage was reporting 0% even if the coverage was ran. Running tox environment in development mode and correctly referencing pynwb path fixed this issue. 

## Checklist

- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR description clearly describes problem and the solution?
- [x] Is your contribution compliant with our coding style ? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number ?

### CODE REVIEW

**Comment by jcfr** [comment]
Hi,

Thanks. Look like the _version.py script wasn't we excluded from the
report. Is the branch on the latest dev ?

May the config setting is ignored ?

(Can't easily check, on my phone)

Jc

On Nov 15, 2017 3:59 PM, "Doruk Ozturk" <notifications@github.com> wrote:

@dorukozturk <https://github.com/dorukozturk> requested your review on:
NeurodataWithoutBorders/pynwb#271
<https://github.com/NeurodataWithoutBorders/pynwb/pull/271> Add usedevelop
flag for coverage-tox issue.

—
You are receiving this because your review was requested.

Reply to this email directly, view it on GitHub
<https://github.com/NeurodataWithoutBorders/pynwb/pull/271#event-1343865534>,
or mute the thread

**Comment by jcfr** [comment]
Not sure why is has to be done in developer mode.

Here is what I used to do to make sure all codecov variables are passed: https://github.com/scikit-build/cmake-python-distributions/blob/a6ed3f1c89a88c3968792d167313716dbb96b33b/.circleci-matrix.yml#L17

May be related is #272

File: .circleci/config.yml [hunk 1]
```diff
@@ -63,21 +63,24 @@ jobs:
...
+     - COVERAGE_TOX_ENV: "py27"
     <<: *ci-steps
 
   python35:
     docker:
       - image: circleci/python:3.5.4-jessie
     environment:
-     - TOX_ENV: "py35,CI-py35"
+     - TEST_TOX_ENV: "py35,CI-py35"
+     - COVERAGE_TOX_ENV: "py35"
```
Comment (jcfr) [inline .circleci/config.yml]: Are the variable switched ? Look like `COVERAGE_TOX_ENV` should be associated with `CI-py35`

Comment (jcfr) [reply .circleci/config.yml]: Also not clear why it is `py35,CI-py35` instead of `CI-py35`

Comment (dorukozturk) [reply .circleci/config.yml]: I think you are right. I think all envs inherit from testenv so CI-py35 should be enough.

Comment (dorukozturk) [reply .circleci/config.yml]: I think I will refactor tox.ini. We should not need comments there.

**Comment by dorukozturk** [comment]
@jcfr I think this is ready. I fixed all the coverage related issues and made appveyor also run tox for consistency. Also fixed the running coverage twice issue and added some comments too.

File: appveyor.yml [hunk 2]
```diff
@@ -9,21 +9,44 @@ environment:
     - PYTHON_VERSION: 2.7
       PYTHON: C:\Miniconda-x64
       ENV: miniconda
+      TEST_TOX_ENV: py27
+      COVERAGE_TOX_ENV: coverage-py27
+      BUILD_TOX_ENV: build-py27
```
Comment (jcfr) [inline appveyor.yml:14]: How do you ensure that tox will run into the miniconda env ? 

References:
* https://stackoverflow.com/questions/30555943/is-it-possible-to-use-tox-with-conda-based-python-installations
* https://github.com/tox-dev/tox/tree/master/doc/drafts and also `https://github.com/tox-dev/tox/issues/273`

**Comment by jcfr** [comment]
Nice, much cleaner. Also nice to rename the tox env.

The issue with this approach is that tox is called  multiple times (once for build, once for test wheel install, once for coverage)

On windows, this kills the CI speed (which was not already great). Now at 20 minutes

I still think have one tox env with conditional execution of command would be the way to go.

**Comment by jcfr** [comment]
> The issue with this approach is that tox is called multiple times (once for build, once for test wheel install, 
> once for coverage)

> On windows, this kills the CI speed (which was not already great). Now at 20 minutes

> I still think have one tox env with conditional execution of command would be the way to go.

Having an other look. I suggest we move forward with this and improve in follow up PR.

We may have to revisit the miniconda testing

**Comment by dorukozturk** [comment]
@jcfr I agree. I can work on conda-tox stuff on another PR.

**Review by jcfr — APPROVED** [review]

**Pull request merged**